notebook/notebookapp.py

```diff
@@ -1399,18 +1421,25 @@ def _update_mathjax_config(self, change):
               "sent by the upstream reverse proxy. Necessary if the proxy handles SSL"))
     )
 
+    file_id = Unicode()
+
+    @default('file_id')
+    def _default_file_id(self):
+        int_hash = "%s" % (hash(self.sock) if self.sock else self.port)
```

Comment:
This is redundant or the cast using `str()` is - take your pick.
```suggestion
        return str(hash(self.sock) if self.sock else self.port)
```